### PR TITLE
Adds missing and improves existing mech attack logging

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
Adds missing and improves existing attack logs for mechs

- alien attack added
- animal attack improved
- Hulk attack is improved
- Throwing added
- projectiles added
- Attackby (items) added

Also makes it so that attacks against non mobs will only be send to admins with ATKLOG_ALL on

## Why It's Good For The Game
Improves the existing logging and adds missing logging so admins can properly solve mech cases

## Images of changes
image

## Changelog
:cl:
add: Alien attacks on mechs are logged
tweak: Animal attacks on mechs are improved
tweak: Hulk attacks on mechs are improved
add: Throwing attacks on mechs are logged
add: Projectiles hitting mechs are logged
add: Attacking mechs with items is now logged
tweak: Attacks against non mobs will now only be send to admins with ATKLOG_ALL on
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
`code/__HELPERS/mobs.dm:293:warning: user_info: variable defined but not used`

**Comment by farie82** [comment]
> `code/__HELPERS/mobs.dm:293:warning: user_info: variable defined but not used`

Yep forgot to save the file after I made the last changes... impressive. Thanks

**Comment by MarsM0nd** [comment]
Does this silence attack logs that didn't cause damage/happened on lavaland? Currently, legion heads do spam the log against a ripley quite a bit, even though it's lavaland and not harming the mech.

**Comment by farie82** [comment]
> Does this silence attack logs that didn't cause damage/happened on lavaland? Currently, legion heads do spam the log against a ripley quite a bit, even though it's lavaland and not harming the mech.

Danke. You just found a bug with the system where hitting atoms which are not mobs send a message to all admins with ATKLOG_MOST on. I'll update the PR to fix that and to not generate logs if no damage is done

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -584,10 +587,14 @@
 
 /obj/mecha/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum) //wrapper
 	log_message("Hit by [AM].")
+	if(isitem(AM))
```
Comment (Fox-McCloud) [inline code/game/mecha/mecha.dm:590]: more than items can be thrown.

Comment (farie82) [reply code/game/mecha/mecha.dm:590]: Yeah I know. Thing is the whole throwing code is fucked up and throwingdatum is not passed.
I didn't want to go rework that entire system for now.
I did a quick analysis and a lot of stuff don't expect throwingdatum. A lot of ways in don't pass throwingdatum and so on. It's a mess

**Pull request merged**